docs/dev/alternative_release_image_sources.md

```diff
@@ -0,0 +1,75 @@
+# Alternative Release-Image Sources
+
+## release-image content vs release-image source
+
+`release-image` content is what operators get installed and what version of each operator (i.e. the container image) gets installed to the cluster. While, `release-image` source is **where** the `release-image` content gets pulled.
+
+Currently the installer controls **both** the `release-image` content and the `release-image` source using the embedded release-image location or the `OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE` env, but to allow users to use a release-image from their private registry (change the `release-image` source), but keep the `release-image` content identical to one vetted by OpenShift, the installer needs to allow specifying source separate from the content.
+
+## Controlling the content
+
+The content of the `release-image`, i.e. the digest, continues to the controlled by the embedded release-image location or the `OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE` env.
+
+## Controlling the source
+
+The installer allows the users to specify a list of `ReleaseImageSources` through the InstallConfig, which become the **authoritative** list of container image repositories that can be used to fetch the `release-image` content which includes the release-image and all the included image references for the various container images for operators.
+
+## Details
+
+### ImageContentSourcePolicy
+
+If a list of `ReleaseImageSources` is specified, the list of digest based [RepositoryDigestMirrors][repository-digest-mirrors] includes,
+
+* the embedded release-image `repository`, and
+* all the items in the `ReleaseImageSources`
+
+**IMPORTANT**: there's an assumption that all the containers image references in the release-image are on the same repository for the embedded release-image.
+
+### release-image location
+
+The release-image location that is propagated to the bootstrap node and the cluster-version-operator will continue to be the embedded release-image location if **NO** `ReleaseImageSources` were specified.
+
+If a list of `ReleaseImageSources` is specified, the release-image location that is propagated to the bootstrap node and the cluster-version-operator will comprise of the repository from the list (preferentially the first one), and the content (i.e the digest) from the embedded release-image.
+
+### Bootstrap machine containers-registries.conf
+
+If a list of `ReleaseImageSources` is specified, the [Registries][registry-containers-registries-conf] will be configured to have each source be a mirror for another.
```

Comment:
Should we consider calling this `ReleaseImageSourceRepositories` or is that too wordy?